### PR TITLE
Migrate persistent_tool_state to null safety

### PR DESCRIPTION
**Pull request by jmagman**
Part of #71511

### CODE REVIEW

File: packages/flutter_tools/lib/src/persistent_tool_state.dart [hunk 1]
```diff
@@ -96,55 +94,59 @@ class _DefaultPersistentToolState implements PersistentToolState {
...
   }
 
   @override
-  String lastActiveVersion(Channel channel) {
-    final String versionKey = _versionKeyFor(channel);
+  String? lastActiveVersion(Channel channel) {
+    final String? versionKey = _versionKeyFor(channel);
     assert(versionKey != null);
-    return _config.getValue(versionKey) as String;
+    return _config.getValue(versionKey!) as String?;
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/persistent_tool_state.dart:105]: Is there any way to mark that the getter returns a nullable but the setter requires a nonnull value?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/persistent_tool_state.dart:105]: I think that sort of thing is frowned upon.

Comment (jmagman) [reply packages/flutter_tools/lib/src/persistent_tool_state.dart:105]: I was frowning the whole time.

Comment (jmagman) [reply packages/flutter_tools/lib/src/persistent_tool_state.dart]: Now the getters and setters are named something slightly different.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**